In [21]:
import pandas as pd # type: ignore
import numpy as np # type: ignore

print_heads = np.array([[1.13], [1.55], [1.43], [0.92], [1.25], [1.36], [1.32], 
              [0.85], [1.07], [1.48], [1.20], [1.33], [1.18], [1.22], [1.29]])
df = pd.DataFrame(print_heads)
df.rename(columns = {0: 'Durability_in_millions'}, inplace = True)
df

Durability_in_millions
0                     1.13
1                     1.55
2                     1.43
3                     0.92
4                     1.25
5                     1.36
6                     1.32
7                     0.85
8                     1.07
9                     1.48
10                    1.20
11                    1.33
12                    1.18
13                    1.22
14                    1.29

In [25]:
df['Durability_in_millions'].mean().round(4)

1.2387

In [38]:
# 1. Finding the 99% confidence interval for the mean number of characters printed before
#    the print-head fails using the sample standard deviation.

import pandas as pd # type: ignore
from scipy import stats # type: ignore

# Given data
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

data_series = pd.Series(data)

# Calculate sample mean and standard deviation
sample_mean = data_series.mean()
sample_std = data_series.std(ddof=1)

# Determine the t-score for a 99% confidence level
t_score = stats.t.ppf(0.995, df=len(data)-1) # two-tailed test

# Calculate margin of error
margin_of_error = t_score * (sample_std / (len(data) ** 0.5))

# Confidence interval
confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)

confidence_interval


(1.090197338451367, 1.3871359948819662)

The 99% confidence interval for the mean number of characters using sample standard deviation is (1.09, 1.387)

In [40]:
# If it were known that the population standard deviation is 0.2 million characters, 
# Construct a 99% confidence interval for the mean number of characters printed before failure.

import pandas as pd # type: ignore
from scipy.stats import norm # type: ignore

# Given data and known population standard deviation
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
population_std = 0.2

# Create a pandas Series
data_series = pd.Series(data)

# Calculate sample mean
sample_mean = data_series.mean()

# Determine z-score for a 99% confidence level
z_score = norm.ppf(0.995) # two-tailed test

# Calculate margin of error using population standard deviation
margin_of_error = z_score * (population_std / (len(data) ** 0.5))

# Construct confidence interval
confidence_interval_population_std = (sample_mean - margin_of_error, sample_mean + margin_of_error)

confidence_interval_population_std


(1.1056514133957607, 1.3716819199375725)

A 99% confidence interval for the mean number of characters printed before failure using 
the population standard deviation is (1.105, 1.37)